uptane/clients/primary.py

```diff
@@ -823,7 +823,7 @@ def get_last_timeserver_attestation(self):
           most_recent_attestation)
       return most_recent_attestation
 
-    else:
+    else: # pragma: no cover
```

Comment:
As to why there isn't a comment in the code: I thought about it, but I hesitated because it's such a meta point, relating to the performance and coverage of our unit tests rather than the actual code here.... What do you think?

As for why this is excluded from coverage:
Yes, there's a reason provided in the PR summary: `"excluded a handful of lines that are clearly forward-defensive (generally inaccessible at present but intended to alert (fail early and loudly) if there are errors in future changes to the code) from code coverage."`  This is such a clause: inaccessible at present (you can't really get this far into the code if the METADATA_FORMAT setting isn't 'json' or 'der' unless you artificially modify it in the middle of operation, in which case there are probably larger problems). The purpose of the clause here is draw developer attention to this section of the code if additional metadata formats are added in future modifications to the project.

Reply:
I think it might be good to have a comment above the else clause that includes the explanation in PR summary.  It makes it clear that its defensive code, and not expected to be executed (and thus should be excluded from coverage).

An example used in the TUF codebase:

```
tuf.formats.METADATA_EXTENSION.check_match(metadata_filename)

if metadata_filename.endswith('json'):
  do something
  ...

# The format check above should have verified that metadata contains a supported extension.
else
  raise Error('Bad metadata extension')
```

Reply:
Fixed, thanks